### DISCUSSION TITLE
thread/suo

### DISCUSSION THREAD

**Discussion by giscus** [itwanger/tobebetterjavaer-giscus#225] [Announcements]
# thread/suo

Java提供了种类丰富的锁，每种锁因其特性的不同，在适当的场景下能够展现出非常高的效率。本文旨在对锁相关源码、使用场景进行举例，为读者介绍主流锁的知识点，以及不同的锁的适用场景。

https://tobebetterjavaer.com/thread/suo.html

**Comment by lichade-son** [comment]
再进入hasQueuedPredecessors()，可以看到该方法主要做一件事情：主要是判断当前线程是否位于同步队列中的第一个。如果是则返回true，否则返回false。

这个是否有点问题，这个好像返回的是当前线程是否有前驱，有前驱的话返回true，感觉跟文中表述相反？
